msteams-platform/concepts/build-and-test/shared-channels.md

```diff
@@ -73,12 +96,75 @@ You can get direct shared channel membership by using the `hostTeamGroupID` from
+
+This subscription enables apps to monitor membership changes in shared channels and its associated teams. For more information on how to create a Microsoft Graph change notification subscription, see [Create a subscription.](/graph/teams-changenotifications-teammembership)
+
+### Manage indirect membership in shared channels
+
+You can manage indirect membership in shared channels using the following Microsoft Graph APIs:
+
+* Use [allMembers](/graph/api/channel-list-allmembers?branch=main&branchFallbackFrom=pr-en-us-13010&view=graph-rest-1.0&tabs=http&preserve-view=true) API to retrieve all users who are members of a specific channel.
+
+    ```http
+    GET /teams/{team-id}/channels/{channel-id}/allMembers
+    ```
+
+* Use [sharedWithTeams](/graph/api/sharedwithchannelteaminfo-list?branch=main&branchFallbackFrom=pr-en-us-13010&view=graph-rest-1.0&tabs=http&preserve-view=true) API to list all teams a channel is shared with.
 
     ```http
-    GET /teams/{host-team-group-id}/channels/{channel-id}/sharedWithTeams/{teamX}/members
+    GET /teams/{team-id}/channels/{channel-id}/sharedWithTeams
     ```
 
+* Use the [allowedMembers](/graph/api/sharedwithchannelteaminfo-list-allowedmembers?branch=main&branchFallbackFrom=pr-en-us-13010&view=graph-rest-1.0&tabs=http&preserve-view=true) API to retrieve users from a shared team who can access a shared channel.
+
+    ```http
+    GET /teams/{team-id}/channels/{channel-id}/sharedWithTeams/{sharewithteamsId}/allowedMembers
+    ```
+
+> [!NOTE]
+> `allowedMembers` API returns only newly associated users and doesn't apply to unshared events.
+
+### Validate user access for membership updates
+
+When an app receives a notification for an indirect membership update, it’s important to verify whether the user still has access to the shared channel as the same user might have both direct and indirect membership. For example, if a user is removed from a team that shares a channel, the app should confirm whether the user's access is truly lost. Use the **doesUserHaveAccess** API to determine whether the user still has access to the shared channel.
+
+```http
+GET /doesUserHaveAccessAsync
+```
+
+Refer to [doesUserHaveAccess API](/graph/api/channel-doesuserhaveaccess?view=graph-rest-beta&tabs=http) to learn more about user accesses and relevant permissions.
+
+### Handle bulk membership changes
+
+Teams supresses individual notifications when a channel is shared with a team or when multiple users are removed. This feature reduces notification volume and improves performance.
+
+#### Use sharedWithTeams Subscription for Bulk Membership Changes
+
+To reduce notification overload during membership updates, such as when a shared channel is added to or removed from a team with thousands of members, use the new SharedWithTeams subscription resource:
+
+`/teams/{team-id}/channels/{channel-id}/sharedWithTeams`
```

Comment:
Let's also provide updating the list of all members for a shared channel using /allmembers api after receiving shared with or unshared from a team notification.